py/variable.cpp

```diff
@@ -57,30 +71,36 @@ Variable_dealloc( Variable* self )
 	PyObject_GC_UnTrack( self );
 	Variable_clear( self );
 	self->variable.~Variable();
-	self->ob_type->tp_free( pyobject_cast( self ) );
+	Py_TYPE( self )->tp_free( pyobject_cast( self ) );
 }
 
 
 static PyObject*
 Variable_repr( Variable* self )
 {
-	return PyString_FromString( self->variable.name().c_str() );
+	return FROM_STRING( self->variable.name().c_str() );
 }
 
 
 static PyObject*
 Variable_name( Variable* self )
 {
-	return PyString_FromString( self->variable.name().c_str() );
+	return FROM_STRING( self->variable.name().c_str() );
 }
 
 
 static PyObject*
 Variable_setName( Variable* self, PyObject* pystr )
 {
+#if PY_MAJOR_VERSION >= 3
+	if( !PyUnicode_Check( pystr ) )
+		return py_expected_type_fail( pystr, "unicode" );
+	self->variable.setName( _PyUnicode_AsString( pystr ) );
```

Comment:
I think `PyUnicode_AsUTF8` should be used instead of the deprecated typedef:
https://github.com/python/cpython/blob/master/Include/unicodeobject.h#L1156

But i'm not sure when that was added. I'm also not worried about support early Py3 versions.